File: v1/src/main/java/com/api/v1/controllers/RentTerminationController.java
package com.api.v1.controllers;

import com.api.v1.services.rent.RentTerminationService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.*;
import reactor.core.publisher.Mono;

@RestController
@RequestMapping("api/v1/rents")
public class RentTerminationController {

    @Autowired
    private RentTerminationService service;

    @PatchMapping("{orderNumber}")
    @ResponseStatus(value = HttpStatus.NO_CONTENT)
    public Mono<Void> terminate(@PathVariable String orderNumber) {
        return service.terminate(orderNumber);
    }

}

File: kotlin/v2/src/main/kotlin/com/api/v2/rent/controllers/RentTerminationController.kt
package com.api.v2.rent.controllers

import com.api.v2.rent.service.RentTerminationService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.PatchMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.ResponseStatus
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("api/v2/rents")
class RentTerminationController {

    @Autowired
    private lateinit var service: RentTerminationService

    @ResponseStatus(value = HttpStatus.OK)
    @PatchMapping("{orderNumber}")
    suspend fun terminate(@PathVariable orderNumber: String) {
        return service.terminate(orderNumber)
    }

}